<?xml version="1.0" encoding="utf-8"?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from driller_with_qr_drill.xacro    | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="driller">
  <link name="base_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://gerard_bauzil/meshes/driller/BD_drill_2.stl"/>
      </geometry>
      <material name="Red">
        <color rgba="0.9 0.1 0.1 1.0"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://gerard_bauzil/meshes/driller/BD_drill_2.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.001"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0.001" ixy="0" ixz="0" iyy="0.001" iyz="0" izz="0.001"/>
    </inertial>
  </link>
  <link name="drill_link">
    <visual>
      <geometry>
        <cylinder length="0.074" radius="0.005"/>
      </geometry>
      <material name="blue">
        <color rgba="0.1 0.1 0.9 1"/>
      </material>
    </visual>
    <collision>
      <geometry name="drill">
        <cylinder length="0.074" radius="0.005"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.001"/>
      <inertia ixx="0.001" ixy="0" ixz="0" iyy="0.001" iyz="0" izz="0.001"/>
    </inertial>
    <collision_checking>
      <!--- This tells to pinocchio to replace the cylinder called "drill" by
             a capsule with the same radius and length -->
      <capsule name="drill"/>
    </collision_checking>
  </link>
  <link name="tag_support_link_top">
    <visual>
      <geometry>
        <mesh filename="package://gerard_bauzil/meshes/driller/qr_drill_top.stl"/>
      </geometry>
      <material name="blue">
        <color rgba="0.1 0.1 0.9 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://gerard_bauzil/meshes/driller/qr_drill_top.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.001"/>
      <inertia ixx="0.001" ixy="0" ixz="0" iyy="0.001" iyz="0" izz="0.001"/>
    </inertial>
  </link>
  <link name="tag_support_link_back">
    <visual>
      <geometry>
        <mesh filename="package://gerard_bauzil/meshes/driller/qr_drill_back.stl"/>
      </geometry>
      <material name="blue">
        <color rgba="0.1 0.1 0.9 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://gerard_bauzil/meshes/driller/qr_drill_back.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.001"/>
      <inertia ixx="0.001" ixy="0" ixz="0" iyy="0.001" iyz="0" izz="0.001"/>
    </inertial>
  </link>
  <joint name="tag_support_joint" type="fixed">
    <parent link="drill_link"/>
    <child link="tag_support_link_top"/>
    <!-- 74mm / 2 + 4mm -->
    <origin rpy="3.141592653589793 0 -1.57079632679" xyz="-0.06 0.00 -0.105"/>
    <!-- <origin xyz="0 0.049 0.041" rpy="0 3.141592653589793 0" /> -->
  </joint>
  <joint name="tag_support_joint_back" type="fixed">
    <parent link="drill_link"/>
    <child link="tag_support_link_back"/>
    <!-- 74mm / 2 + 4mm -->
    <origin rpy="3.141592653589793 0 -1.57079632679" xyz="-0.0 0.00 -0.194"/>
    <!-- <origin xyz="0 0.049 0.041" rpy="0 3.141592653589793 0" /> -->
  </joint>
  <link name="tag36_11_00023">
    <visual>
      <origin xyz="0 0 0.00001"/>
      <geometry>
        <mesh filename="package://gerard_bauzil/meshes/apriltag_36h11/tag36_11_00023.dae" scale="0.04 0.04 1."/>
      </geometry>
    </visual>
  

  </link>
  <link name="tag36_11_00230">
    <visual>
      <origin xyz="0 0 0.00001"/>
      <geometry>
        <mesh filename="package://gerard_bauzil/meshes/apriltag_36h11/tag36_11_00230.dae" scale="0.04 0.04 1."/>
      </geometry>
    </visual>
       
  </link>
  <joint name="tag_joint" type="fixed">
    <parent link="tag_support_link_top"/>
    <child link="tag36_11_00023"/>
    <origin xyz="0 0 0.002"/>
  </joint>
  <joint name="tag_joint_back" type="fixed">
    <parent link="tag_support_link_back"/>
    <child link="tag36_11_00230"/>
    <origin xyz="0 0 0.002"/>
  </joint>
  <!-- <xacro:qr_drill number="_00023" size="0.04" /> -->
  <!-- The drill penetrates in the driller of 74 - 46.5 = 27.5mm -->
  <joint name="drill_joint" type="fixed">
    <origin rpy="0 1.5707963267948966 0" xyz="0.121 0 0.074"/>
    <parent link="base_link"/>
    <child link="drill_link"/>
  </joint>
  <link name="support_link"/>
  <joint name="support_joint" type="fixed">
    <origin rpy="0 0.07256862 0" xyz="0 0 -0.1125"/>
    <parent link="base_link"/>
    <child link="support_link"/>
  </joint>
</robot>

